app/views/layouts/_market_nav.html.erb

```diff
@@ -0,0 +1,5 @@
+<%= link_to 'All Markets', markets_path %>
+
+<%= link_to 'Market', market_path(@market) if params[:market_id] %>
+
+<%= link_to 'Logout', :root %>
```

Comment:
We went back and forth on location of these files, since market and vendor are both models and types of users in our session-less authentication system. I like the idea of putting it in the shared folder.